src/rust/ensogl/src/system/gpu/data/texture/class.rs

```diff
@@ -45,6 +45,109 @@ impl Drop for TextureBindGuard {
 
 
 
+// ==================
+// === Parameters ===
+// ==================
+
+/// Helper struct to specify texture parameters that need to be set when binding a texture.
+///
+/// The essential parameters that need to be set are about how the texture will be samples, i.e.,
+/// how the values of the texture are interpolated at various resolutions, and how out of bounds
+/// samples are handled.
+///
+/// For more background see: https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/texParameter
+///
+#[derive(Copy,Clone,Debug,Default)]
+pub struct Parameters {
+    /// Specifies the setting for the texture magnification filter (`Context::TEXTURE_MIN_FILTER`).
+    pub min_filter : MinFilterValue,
```

Comment:
Small naming issue - we try not to add suffixes like `Value` or `Trait` to our names.